src/integrationTest/java/com/mongodb/hibernate/embeddable/StructAggregateEmbeddableIntegrationTests.java

```diff
@@ -114,38 +160,314 @@ void testNestedValues() {
+                }
+                """);
+        var loadedItem = sessionFactoryScope.fromTransaction(
+                session -> session.find(ItemWithNestedValueHavingArraysAndCollections.class, item.id));
+        assertEq(item, loadedItem);
+    }
+
+    /**
+     * This test also covers the behavior of an empty {@linkplain Struct struct} aggregate {@linkplain Embeddable
+     * embeddable} value, that is one having {@code null} as the value of each of its persistent attributes.
+     *
+     * @see EmbeddableIntegrationTests#testFlattenedValueHavingNullArraysAndCollections()
+     * @see ArrayAndCollectionIntegrationTests#testArrayAndCollectionValuesOfEmptyStructAggregateEmbeddables()
+     */
+    @Test
+    @Disabled("TODO-HIBERNATE-48 https://jira.mongodb.org/browse/HIBERNATE-48 enable this test")
+    public void testNestedValueHavingNullArraysAndCollections() {
+        var emptyStructAggregateEmbeddable = new ArraysAndCollections(
+                null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, null,
+                null, null, null, null, null, null, null);
+        var item = new ItemWithNestedValueHavingArraysAndCollections(1, emptyStructAggregateEmbeddable);
+        sessionFactoryScope.inTransaction(session -> session.persist(item));
+        assertCollectionContainsExactly(
+                """
+                {
+                    _id: 1,
+                    nested: {
+                        bytes: null,
+                        chars: null,
+                        ints: null,
+                        longs: null,
+                        doubles: null,
+                        booleans: null,
+                        boxedChars: null,
+                        boxedInts: null,
+                        boxedLongs: null,
+                        boxedDoubles: null,
+                        boxedBooleans: null,
+                        strings: null,
+                        bigDecimals: null,
+                        objectIds: null,
+                        structAggregateEmbeddables: null,
+                        charsCollection: null,
+                        intsCollection: null,
+                        longsCollection: null,
+                        doublesCollection: null,
+                        booleansCollection: null,
+                        stringsCollection: null,
+                        bigDecimalsCollection: null,
+                        objectIdsCollection: null,
+                        structAggregateEmbeddablesCollection: null
+                    }
+                }
+                """);
+        var loadedItem = sessionFactoryScope.fromTransaction(
+                session -> session.find(ItemWithNestedValueHavingArraysAndCollections.class, item.id));
+        // `loadedItem.nested` is `null` despite `item.nested` not being `null`.
+        // TODO-HIBERNATE-48 There is nothing we can do here, such is the Hibernate ORM behavior. - or can we?
```

Comment:
If Hibernate decided to return null, we'd better accept the truth. There is no official Hibernate extension point since v7. We can't change the behaviour without changing the class file (e.g. AOP) which is out of our considersation currently.

There is a Hibernate bug that empty struct won't end up with null when its containing array is loaded. I created the ticket at after confirmation: https://hibernate.atlassian.net/browse/HHH-19575